src/modules/retransmitter/retransmitter.cpp

```diff
@@ -99,12 +99,24 @@ static void wait_for_msg() {
         }
     }
 }
-extern "C" void task_retransmitter_fn(void) {
 
-    ros_echronos::NodeHandle nh;
+ros_echronos::Publisher<std_msgs::Float64> * publishers[NUM_MSG];
+std_msgs::Float64 msg;
+struct messageAdapter serial;
+ros_echronos::NodeHandle nh;

```

Comment:
There's a bug in how we've setup C++ and global objects don't get initialised correctly. `nh` and `msg` need to be pointers that get initialised in one of the task functions (task functions never exit so it's safe to share pointers to things declared on the stack in their scope).

Reply:
`msg` and `nh` are now being initialised within the first task `task_retransmitter_fn`